src/parser/jobs/gnb/modules/NoMercy.tsx

```diff
@@ -37,228 +34,87 @@ const EXPECTED_USES = {
-			const fightTimeRemaining = this.parser.fight.end_time - event.timestamp
-			noMercyState.isRushing = NO_MERCY_BUFF_DURATION >= fightTimeRemaining
+			{
+				action: ACTIONS.SONIC_BREAK,
+				expectedPerWindow: EXPECTED_USES.SONIC_BREAK,
+			},
 
-			this.noMercyWindows.push(noMercyState)
-		}
+			{
+				action: ACTIONS.BLASTING_ZONE,
+				expectedPerWindow: EXPECTED_USES.BLASTING_ZONE,
+			},
 
-		// So long as we're in this window, log our actions to it.
-		const lastNoMercy = this.lastNoMercy
-		if (lastNoMercy != null && lastNoMercy.end == null) {
-			lastNoMercy.rotation.push(event)
+			{
+				action: ACTIONS.BOW_SHOCK,
+				expectedPerWindow: EXPECTED_USES.BOW_SHOCK,
+			},
 
-			const action = getDataBy(ACTIONS, 'id', actionId) as TODO
-			if (!action) { return }
+			{
+				action: ACTIONS.ROUGH_DIVIDE,
+				expectedPerWindow: EXPECTED_USES.ROUGH_DIVIDE,
+			},
+		],
 
-			// Pre-process on the number of certain things we did
-			if (action.onGcd) {
-				lastNoMercy.numGcds++
-			}
+		suggestionContent: <Trans id="gnb.nomercy.suggestions.expected-uses.content">
+			Watch your uses of certain abilities during <ActionLink {...ACTIONS.NO_MERCY} />. Under ideal conditions, you should
+				be using <ActionLink {...ACTIONS.SONIC_BREAK} />, a full <ActionLink {...ACTIONS.GNASHING_FANG} /> combo, and all of
+				your off-GCD skills - <ActionLink {...ACTIONS.BLASTING_ZONE} />, <ActionLink {...ACTIONS.BOW_SHOCK} />, and at least one
+				charge of <ActionLink {...ACTIONS.ROUGH_DIVIDE} /> - under the buff duration.
+		</Trans>,
+		severityTiers: SEVERITIES.MISSING_EXPECTED_USES,
 
-			switch (actionId) {
-			// Blasting Zone is just a traited version of Danger Zone
-			// Impossible to see both in the same log outside of hacking
-			case ACTIONS.BLASTING_ZONE.id:
-			case ACTIONS.DANGER_ZONE.id:
-				lastNoMercy.numBlastingZones++
-				break
-			case ACTIONS.SONIC_BREAK.id:
-				lastNoMercy.numSonicBreaks++
-				break
-			case ACTIONS.ROUGH_DIVIDE.id:
-				lastNoMercy.numRoughDivides++
-				break
-			case ACTIONS.GNASHING_FANG.id:
-				lastNoMercy.numGnashingFangs++
-				break
-			case ACTIONS.BOW_SHOCK.id:
-				lastNoMercy.numBowShocks++
-				break
-			}
-		}
 	}
 
-	private onRemoveNoMercy(event: BuffEvent) {
-		const lastNoMercy = this.lastNoMercy
+	changeExpectedTrackedActionClassLogic(buffWindow: BuffWindowState, action: BuffWindowTrackedAction) {
 
-		if (lastNoMercy != null) {
-			lastNoMercy.end = event.timestamp
-		}
-	}
+		if (action.action.id === ACTIONS.BURST_STRIKE.id) {
 
-	private onComplete() {
-		// Exclude any window at the end of a fight which could not possibly have gotten all of its GCDs
-		const missedGcds = this.noMercyWindows
-			.filter(window => !window.isRushing)
-			.reduce((sum, window) => sum + Math.max(0, EXPECTED_USES.GCD - window.numGcds), 0)
+			if (buffWindow.rotation.find(cast => cast.ability.guid === ACTIONS.BLOODFEST.id) && (this.getBaselineExpectedTrackedAction(buffWindow, action) === 2)) {
```

Comment:
Could you break this up a bit? It's longer than me blogging for an hour in discord.

Reply:
I pressed the enter key